Standards/scs-0112-v1-volume-type-standard.md

```diff
@@ -0,0 +1,227 @@
+| d34b5de0-d7bf-43cf-8a74-3827169d6616 | LUKS        | True      | cipher='aes-xts-plain64', control_location='front-end', encryption_id='217386bc-1e9b-46a3-9e0e-3ad62c07826c', key_size='256', provider='nova.volume.encryptors.luks.LuksEncryptor' |
++--------------------------------------+-------------+-----------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+As this is not visible for users, it will be necessary that deployers state that the volume type is encrypted in the description of the volume type in the following way:
+The description needs to begin with `[encrypted]`, after that any further description is allowed. It should look like this example:
+
+```text
+openstack volume type show LUKS
++--------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| Field              | Value                                                                                                                                                        |
++--------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| access_project_ids | None                                                                                                                                                         |
+| description        | [encrypted] This volume uses LUKS-encryption                                                                                                                 |
+| id                 | d63307fb-167a-4aa0-9066-66595ea9fb21                                                                                                                         |
+| is_public          | True                                                                                                                                                         |
+| name               | LUKS                                                                                                                                                         |
+| qos_specs_id       | None                                                                                                                                                         |
++--------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+### Replication
+
+Replication states, whether or not there are multiple replicas of a volume. Thus answers the question, whether the data could survive a node outage. Unfortunately there are two ways replication can be achieved:
```

Comment:
In line 79, you define replicated volumes MUST be replicated three times. Here,  you dilute this requirement. This standard must be consistent and I recommend to add the number of replicas here as well.

Reply:
as was discussed during a meeting regarding this standard:

There is currently no way to discover the replication factor used by the most used backend (ceph).

I believe this is even mentioned somewhere in the prelude of the standard, at least in some earlier iterations it was there.

Regardless, given the above facts, I don't think it makes sense to standardize a certain number of replicas, given that we can't test this.

Reply:
line 79 does only provide examples of possible volume types. Maybe this should be stated more clearly.

The number of replicas is difficult to get, because there are two ways replication can be achieved that exclude each other. 

1. Replication can be set in the volume type. There you can see the information that the volume will be replicated and the Number of Replicas OpenStack will set.
2. Performancewise it is better to let replication happen in the backend itself and not having the need of OpenStack caring about it. Ceph e.g. does it on its own. It is set in the configuration of ceph (https://docs.ceph.com/en/latest/rados/operations/pools/), where also the number of replicas can be set on a per pool base. So it is not possible for OpenStack to get the information (at least for now, but I do think this will be really hard to get into the volume type database of OpenStack.). The way we do have right now, is that the operator creating or editing the volume type with the ceph backend, will write down the information in the description field.

In conclusion I think the number of replicas would be really nice to have, but maybe will be too hard to get at all.